app/screens/edit_channel/edit_channel.js

```diff
@@ -0,0 +1,297 @@
+                    this.close();
+                });
+                break;
+            case RequestStatus.FAILURE:
+                this.emitUpdating(false);
+                this.setState({error: updateChannelRequest.error, updating: false});
+                break;
+            }
+        }
+    }
+
+    close = () => {
+        this.props.navigator.pop({animated: true});
+    };
+
+    emitCanUpdateChannel = (enabled) => {
+        const buttons = {
+            rightButtons: [{...this.rightButton, disabled: !enabled}]
+        };
+
+        this.props.navigator.setButtons(buttons);
+    };
+
+    emitUpdating = (loading) => {
+        const buttons = {
+            rightButtons: [{...this.rightButton, disabled: loading}]
+        };
+
+        this.props.navigator.setButtons(buttons);
+    };
+
+    validateDisplayName = (displayName) => {
+        const {formatMessage} = this.context.intl;
+
+        if (!displayName) {
+            return {error: formatMessage(messages.display_name_required)};
+        } else if (displayName.length > ViewTypes.MAX_CHANNELNAME_LENGTH) {
+            return {error: formatMessage(
+                messages.display_name_maxLength,
+                {maxLength: ViewTypes.MAX_CHANNELNAME_LENGTH}
+            )};
+        } else if (displayName.length < ViewTypes.MIN_CHANNELNAME_LENGTH) {
+            return {error: formatMessage(
+                messages.display_name_minLength,
+                {minLength: ViewTypes.MIN_CHANNELNAME_LENGTH}
+            )};
+        }
+
+        return {error: null};
+    };
+
+    validateChannelURL = (channelURL) => {
+        const {formatMessage} = this.context.intl;
+
+        if (!channelURL) {
+            return {error: formatMessage(messages.name_required)};
+        } else if (channelURL.length > ViewTypes.MAX_CHANNELNAME_LENGTH) {
+            return {error: formatMessage(
+                messages.name_maxLength,
+                {maxLength: ViewTypes.MAX_CHANNELNAME_LENGTH}
+            )};
+        }
+
+        const cleanedName = cleanUpUrlable(channelURL);
+        if (cleanedName === channelURL) {
+            return {error: null};
+        }
+
+        return {error: formatMessage(messages.name_lowercase)};
+    };
+
+    onUpdateChannel = async () => {
+        Keyboard.dismiss();
+        const {displayName, channelURL, purpose, header} = this.state;
+        const {channel: {id}} = this.props;
+        const channel = {
+            display_name: displayName,
+            name: channelURL,
+            purpose,
+            header
+        };
+
+        let result = this.validateDisplayName(displayName.trim());
+        if (result.error) {
+            this.setState({error: result.error});
+            return;
+        }
+        result = this.validateChannelURL(channelURL.trim());
+        if (result.error) {
+            this.setState({error: result.error});
+            return;
+        }
+
+        alertErrorIfInvalidPermissions(
```

Comment:
by looking at the code you don't need this as you are already showing an error in an <ErrorText> component if the server returns one, so no need to show the alert unless you prefer that and remove the other error handling, I would keep the one you have and remove this